### ISSUE TITLE
sql: investigate Hibernate's  org.hibernate.jpa.test.criteria.fetchscroll.CriteriaToScrollableResultsFetchTest testWithScroll

### ISSUE THREAD

**Issue by BramGruneir** [cockroachdb/cockroach#26735]
While looking at Hibernate's test suite, the following test failed:
org.hibernate.jpa.test.criteria.fetchscroll.CriteriaToScrollableResultsFetchTest testWithScroll

The code that sets this up
```java
Query hibernateQuery = query.unwrap( Query.class );
hibernateQuery.setReadOnly( true );
hibernateQuery.setCacheable( false );

List<OrderLine> lines = new ArrayList<>();
ScrollableResults scrollableResults = hibernateQuery.scroll();
scrollableResults.last();
int rows = scrollableResults.getRowNumber() + 1;
scrollableResults.beforeFirst();
while ( scrollableResults.next() ) {
  lines.add( (OrderLine) scrollableResults.get( 0 ) );
}
```

Operation requires a scrollable ResultSet, but this ResultSet is FORWARD_ONLY.

This will require some investigation using a tool such as [Wireshark](https://www.wireshark.org/) to compare Cockroach and Postgres' response.

**Comment by knz** [comment]
I suspect this requires SQL cursors, which CockroachDB does not support. (and probably does not aim to support). But indeed we still need to check the low level queries.

**Comment by knz** [comment]
@BramGruneir This issue is improperly qualified. It does neither state what is the deviation (difference between expected behavior and actual behavior) nor what action items should be carried out.

Unless you provide either (preferably both), we can't do anything with this.

**Comment by BramGruneir** [comment]
I didn't see any cursors in the SQL, but some more time investigating the failure is needed as that might be correct.

That being said, I think the expected and actual behaviour is in the top level of the issue.

Hibernate expects that a resultset can be scrollable both forward and backward, but we only support forward.

The actionable item is to investigate this further with a tool such as Wireshark.

**Comment by knz** [comment]
Define:

- what is a "resultset" for hibernate, and what is the correspondance between that concept and SQL
- what does it mean for a "resultset" to be "scrollable". How does Hibernate make that determination in SQL.

Until these definitions are specified, the issue is unactionable. I am requalifying this as investigation.

**Comment by knz** [comment]
The title was wrong really. We do not really know what the issue is. The title must reflect the investigation, not the working theory.

**Comment by knz** [comment]
The doc you link here specifically mentions SQL cursors. The investigation will need to determine whether:

- indeed this requires SQL cursors
- whether a hibernate dialect can provide scrollability by other means, or whether this is owned by the underlying JDBC driver (in this case, postgres').

**Comment by jordanlewis** [comment]
Fixed in 2.1.
